### PR TITLE
Swift2

### PR DESCRIPTION
**Pull request by chenjsa**

### CODE REVIEW

File: Documentation/BasicOperators.md [hunk 1]
```diff
@@ -192,7 +192,10 @@ stream completes.
 
 ```Swift
 let (signal, observer) = Signal<Int, NoError>.pipe()
-signal.collect().observe(next: print)
+
+signal
+    .collect()
+    .observeNext{ next in print(next) }
```
Comment (NachoSoto) [inline Documentation/BasicOperators.md]: Nit: space after `observeNext`

File: Documentation/BasicOperators.md [hunk 2]
```diff
@@ -451,7 +466,9 @@ signal
             return .Other
         }
     }
-    .observe(error: print)
+    .observeError { error in
+        print(error.description)
```
Comment (NachoSoto) [inline Documentation/BasicOperators.md]: `description ` isn't needed here, right?

File: Documentation/BasicOperators.md [hunk 3]
```diff
@@ -413,8 +418,18 @@ let producer = SignalProducer<String, NSError> { (observer, _) in
 producer
     .on(error: {e in print("Error")})             // prints "Error" twice
     .retry(2)
-    .start(next: print,                           // prints "Success"
-          error: { _ in print("Signal Error")})
+    .start{ event in
```
Comment (NachoSoto) [inline Documentation/BasicOperators.md]: You need a space before every `{`

File: Documentation/BasicOperators.md [hunk 4]
```diff
@@ -375,17 +380,17 @@ observerC.sendNext("Z")        // prints "Z"
 
 These operators are used to handle errors that might occur on an event stream.
 
-### Catching errors
+### Flat map errors
```
Comment (NachoSoto) [inline Documentation/BasicOperators.md]: I'd still call this section "Catching errors"

File: Documentation/BasicOperators.md [hunk 5]
```diff
@@ -249,8 +253,9 @@ has sent at least N values.
 let (numbersSignal, numbersObserver) = Signal<Int, NoError>.pipe()
 let (lettersSignal, lettersObserver) = Signal<String, NoError>.pipe()
 
-zip(numbersSignal, lettersSignal)
-    .observe(next: print, completed: { print("Completed") })
+let signal = zip(numbersSignal, lettersSignal)
+signal.observeNext{ next in print("Next: \(next)") }
+signal.observeCompleted{ print("Completed") }
```
Comment (NachoSoto) [inline Documentation/BasicOperators.md]: Here I'd recommend using
```Swift
signal.observe { event in
  switch event {
  ...
  }
}
```

**Comment by NachoSoto** [comment]
Thank you! Just a few nits :star:

File: Documentation/BasicOperators.md [hunk 6]
```diff
@@ -138,7 +138,7 @@ let (signal, observer) = Signal<String, NoError>.pipe()
 
 signal
     .map { string in string.uppercaseString }
-    .observe(next: print)
+    .observeNext{ next in print(next) }
```
Comment (neilpa) [inline Documentation/BasicOperators.md]: Spacing here too

**Comment by chenjsa** [comment]
It's ok now.

**Comment by NachoSoto** [comment]
Thank you!

**Pull request merged**